### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by GitKurmax**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  
+}
```
Comment (gestap) [inline .vscode/launch.json]: add `.vscode` to .gitignore and don't push it to repo

File: script.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+class Todo {
```
Comment (gestap) [inline script.js]: use strict mode

File: script.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    item.priority = priority;
+    this.list.push(item);
+    item.id = this.list.length;
+    return item.id;
+  }
+
+  removeItem(id) {
+    for (let item of this.list) {
+      if (item.id === id) {
+        this.list.splice(this.list(indexOf(item), 1));
```
Comment (gestap) [inline script.js]: try to use `findIndex` in this case it will better and shorter code

File: script.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        return true;
+      }
+      return false;
+    }
+  }
+
+  getItem(id) {
+    for (let item of this.list) {
+      if (item.id === id) {
+        return this.list[this.list.indexOf(item)];
```
Comment (gestap) [inline script.js]: try to use `find` in this case it will better and shorter code

File: script.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        throw new Error("the list is empty");
+      }
+      this.list.sort((a, b) => {
+        if(a.priority > b.priority) return -1;
+        if(a.priority < b.priority) return 1;
+        if(a.priority === b.priority) return 0;
+      });
+      return this.list[0];
+    } catch (e) {
+      return e.message;
```
Comment (gestap) [inline script.js]: you need to `...throw an error` instead return message

**Review by ost-ptk — CHANGES_REQUESTED** [review]

**Review by NclsVI — APPROVED** [review]

**Pull request closed without merging**